File: Java/Counting.java
package Algoritmo;

import java.util.Arrays;

public class Counting {

	public static void main(String[] args) {
		int[] arr = { 12, 11, 13, 5, 6 };
		countingSort(arr);
		for (int i : arr) {
			System.out.print(i + " ");
		}
	}

	public static void countingSort(int[] arr) {
		int max = Arrays.stream(arr).max().getAsInt();
		int min = Arrays.stream(arr).min().getAsInt();
		int range = max - min + 1;
		int[] count = new int[range];
		int[] output = new int[arr.length];

		for (int i = 0; i < arr.length; i++) {
			count[arr[i] - min]++;
		}

		for (int i = 1; i < range; i++) {
			count[i] += count[i - 1];
		}

		for (int i = arr.length - 1; i >= 0; i--) {
			output[count[arr[i] - min] - 1] = arr[i];
			count[arr[i] - min]--;
		}

		System.arraycopy(output, 0, arr, 0, arr.length);
	}
}

File: Python/Counting.py
def counting_sort(arr):
    max_value = max(arr)
    count = [0] * (max_value + 1)

    for num in arr:
        count[num] += 1

    sorted_arr = []
    for i in range(max_value + 1):
        sorted_arr.extend([i] * count[i])

    return sorted_arr

arr = [12, 11, 13, 5, 6]
arr = counting_sort(arr)
print("Array ordenado:")
print(arr)